File: L1_999/7. Reverse Integer.kt
fun reverse(x: Int): Int {
    val multiply = if (x < 0) -1 else 1
    val reverse = x.toString().reversed().removeSuffix("-")
    val result = reverse.toLong() * multiply
    return if (is32Bit(result)) result.toInt() else 0
}

fun is32Bit(num: Long): Boolean {
    val intVal = num.toInt()
    return intVal.toLong() == num
}

File: L1_999/7. Reverse Integer.ts
function reverse(x: number): number {
  let multiply = x > 0 ? 1 : -1
  let reverse = x.toString().split('').reverse().join('')
  let result = Number.parseInt(reverse) * multiply
  return is32Bit(result) ? result : 0
};

function is32Bit(num: number): boolean {
  return (num | 0) === num;
}